File: ARC/ARC050/B.nim
import sequtils, strutils, math

var
    R, B: int
    x, y: int
(R, B) = stdin.readLine.split.map(parseInt)
(x, y) = stdin.readLine.split.map(parseInt)

proc calc(k: int): bool =
    var
        r = (R - k) div (x - 1)
        b = (B - k) div (y - 1)
    if k <= r + b:
        return true
    else:
        return false

proc solve() =
    var
        left = 0
        right = min(R, B) + 1
    while 1 < right - left:
        var mid = (left + right) div 2
        if calc(mid):
            left = mid
        else:
            right = mid
    echo left

solve()

File: ARC/ARC050/B.py
R, B = map(int, input().split())
x, y = map(int, input().split())


def calc(k):
    r = (R - k) // (x - 1)
    b = (B - k) // (y - 1)
    if k <= r + b:
        return True
    else:
        return False


def solve():
    left, right = 0, min(R, B) + 1
    while 1 < right - left:
        mid = (left + right) // 2
        if calc(mid):
            left = mid
        else:
            right = mid
    print(left)


if __name__ == '__main__':
    solve()
